fix(route): change to use api get data and refactor to v2

Fatpandac opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 该 PR 相关 Issue / Involved issue

Close #11553

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/163/ds/9c1c00448e28455781e8fde62830160d
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [X] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [X] 文档说明 Documentation
  - [X] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [X] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [X] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

Fatpandac committed:
fix(route): change to use api get data and refactor to v2

TonyRL commented on lib/v2/netease/ds.js:
```diff
@@ -0,0 +1,36 @@
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id;
+    if (!id) {
+        throw Error('Bad parameter. See <a href="https://docs.rsshub.app/game.html#wang-yi-da-shen">https://docs.rsshub.app/game.html#wang-yi-da-shen</a>');
+    }
```
Unreachable condition because 
https://github.com/DIYgod/RSSHub/blob/1e22bc7a5240e00539d4662663ca1b97b40b3324/lib/v2/netease/router.js#L6
will throw if `id` does not exist

Fatpandac replied:
Yeah, this code is copied from the older one that looks like not necessary, I will remove it.

Fatpandac committed:
fix(route): remove unreachable condition

TonyRL committed:
refactor: change route path

feat!: remove `/open163/latest` since `https://c.open.163.com/open/getLatestMovies.do` is broken

TonyRL committed:
fix: radar
